sites/game-client/ui/overlay.mjs

```diff
@@ -61,8 +79,12 @@ class Overlay {
   */
   onClickToolbarButton() {
     // WIP(fanlan1210)
-    if (this.canHide() && this.position !== undefined) this.hide();
-    else this.show();
+    console.assert(this.positionCache, 'positionCache does not have default value.');
```

Comment:
Optional: I would prefer separating the `positionCache` and the `defaultPosition`.
i.e.
```
if (!this.positionCache) {
  console.assert(this.defaultPosition, 'Derived class should specify this.defaultPosition');
  this.positionCache = this.defaultPosition;
}
```

This way, if something else needs the `defaultPosition`, we'll be able to use it as well.